sqlglot/parser.py

```diff
@@ -4284,11 +4284,16 @@ def _parse_bracket(self, this: t.Optional[exp.Expression]) -> t.Optional[exp.Exp
                 self.expression(exp.Slice, expression=self._parse_conjunction())
             ]
         else:
+            # Temporarily disable the colon range parser (if any) so that we will end up
```

Comment:
don't do this

Reply:
Yeah it's not very good... let me see if I can refactor it

Reply:
any ideas on this?

Reply:
Taking a look now, it's a bit gnarly

Reply:
@tobymao take a look at https://github.com/tobymao/sqlglot/pull/2744/commits/3910a94299027b11f018b84d9df59e6a2c8976c6